x-pack/plugins/ml/server/routes/data_frame_analytics.ts

```diff
@@ -285,12 +335,37 @@ export function dataFrameAnalyticsRoutes({ router, mlLicense }: RouteInitializat
     mlLicense.fullLicenseAPIGuard(async (context, request, response) => {
       try {
         const { analyticsId } = request.params;
+
+        const { destinationIndex } = request.query;
+
+        // If user checks box to deleete the destinationIndex associcated with the job
+        if (destinationIndex) {
+          // Verify if user has privilege to delete the destination index
+
+          const userCanDeleteDestIndex = userCanDeleteIndex(context, destinationIndex);
+          // If user does have privilege to delete the index, then delete the index
+          if (userCanDeleteDestIndex) {
+            await context.ml!.mlClient.callAsCurrentUser('indices.delete', {
+              index: destinationIndex,
+            });
+          }
+        }
+
+        // Delete the dataframe analytics
         const results = await context.ml!.mlClient.callAsCurrentUser(
           'ml.deleteDataFrameAnalytics',
           {
             analyticsId,
           }
         );
+
+        // Delete the index pattern if there's an index pattern that matches the name of dest index
+        const indexPatternId = await getIndexPatternId(context, destinationIndex);
+
+        if (indexPatternId) {
+          const result = deleteIndexPatternById(context, indexPatternId);
```

Comment:
You don't anything with `result`, so a variable definition is redundant. You should get an error from `eslint` about issues like this one 🤔 . Worth to set up a `eslint` plugin for your IDE of choice